src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -1842,6 +1943,41 @@ WorkersContainingAllShards(List *prunedShardIntervalsList)
 }
 
 
+static List *
+WorkersContainingModifyShards(List *prunedShardIntervalsList)
+{
+	ListCell *prunedShardIntervalCell = NULL;
+	List *currentPlacementList = NIL;
+
+	foreach(prunedShardIntervalCell, prunedShardIntervalsList)
+	{
+		List *shardIntervalList = (List *) lfirst(prunedShardIntervalCell);
+		ListCell *shardIntervalCell = NULL;
+
+		if (shardIntervalList == NIL)
+		{
+			continue;
+		}
```

Comment:
This `if` block seems redundant, since the `foreach` below will be a noop if `shardIntervalList` is `NIL` (empty lists are represented as `NIL`).